src/DbalStore.php

```diff
@@ -0,0 +1,303 @@
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getMultiple(array $keys, $default = null)
+    {
+        KeyUtil::validateMultiple($keys);
+
+        // Normalize indices of the array
+        $keys = array_values($keys);
+        $data = $this->doGetMultiple($keys);
+
+        $results = array();
+        $resolved = array();
+        foreach ($data as $row) {
+            $results[$row['meta_key']] = Serializer::unserialize($row['meta_value']);
+            $resolved[$row['meta_key']] = $row['meta_key'];
+        }
+
+        $notResolvedArr = array_diff($keys, $resolved);
+        foreach ($notResolvedArr as $notResolved) {
+            $results[$notResolved] = $default;
+        }
+
+        return $results;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getMultipleOrFail(array $keys)
+    {
+        KeyUtil::validateMultiple($keys);
+
+        // Normalize indices of the array
+        $keys = array_values($keys);
+
+        $data = $this->doGetMultiple($keys);
+
+        $results = array();
+        $resolved = array();
+        foreach ($data as $row) {
+            $results[$row['meta_key']] = Serializer::unserialize($row['meta_value']);
+            $resolved[] = $row['meta_key'];
+        }
+
+        $notResolvedArr = array_diff($keys, $resolved);
+
+        if (!empty($notResolvedArr)) {
+            throw NoSuchKeyException::forKeys($notResolvedArr);
+        }
+
+        return $results;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function remove($key)
+    {
+        KeyUtil::validate($key);
+
+        try {
+            $result = $this->connection->delete($this->table, array('meta_key' => $key));
+        } catch (\Exception $e) {
+            throw WriteException::forException($e);
+        }
+
+        return $result === 1;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function exists($key)
+    {
+        KeyUtil::validate($key);
+
+        try {
+            $result = $this->connection->fetchAssoc('SELECT * FROM '.$this->table.' WHERE meta_key = ?', array($key));
+        } catch (\Exception $e) {
+            throw ReadException::forException($e);
+        }
+
+        return $result ? true : false;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function clear()
+    {
+        try {
+            $connection = $this->connection->query('DELETE FROM '.$this->table);
+            $connection->execute();
+        } catch (\Exception $e) {
+            throw WriteException::forException($e);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function keys()
+    {
+        try {
+            $stmt = $this->connection->query('SELECT meta_key FROM '.$this->table);
+            $result = $stmt->fetchAll(\PDO::FETCH_COLUMN);
+        } catch (\Exception $e) {
+            throw ReadException::forException($e);
+        }
+
+        return $result;
+    }
+
+    public function configureTable()
```

Comment:
I'd keep this as a public method, rename to `createTable()` and drop `configureSchema()` instead (as explained below).